### ISSUE TITLE
Tooltip does not show on condition case

### ISSUE THREAD

**Issue by xyrintech** [ng-bootstrap/ng-bootstrap#2033]
I am using the latest Angular5 version and and latest ng-bootstrap version. 
I am trying to show tooltip conditionally 

```
<input type="text" id="name" class="form-control" [ngbTooltip]="tipContent" placement="bottom" name="name" formControlName="name">
<ng-template *ngIf="true === true" #tipContent>Hello</ng-template>
```

But this does not show up. Seems to have some issue.

**Comment by pkozlowski-opensource** [comment]
Not sure what you mean by "show tooltip conditionally" but what you are trying to do is definitively _not_ the way to go...  What conditions should trigger tooltip showing / hiding? Maybe you should have a look at the manual triggers on our demo page? https://ng-bootstrap.github.io/#/components/tooltip/examples

**Comment by xyrintech** [comment]
Think of condition like form validation.
Now lets say if we need to enable tool tip if form element is not valid.

**Comment by xyrintech** [comment]
I have reviewed the examples. I think this case is not specified in those examples.

**Comment by xyrintech** [comment]
Please suggest if you need aby further details in order to understand

**Comment by xyrintech** [comment]
Any*

**Comment by pkozlowski-opensource** [comment]
> Think of condition like form validation.
Now lets say if we need to enable tool tip if form element is not valid. 

This is a valid use-case but I believe that this is already covered by the basic building blocks we've got in place. What you would have to do is to create a very simple directive that would:
- inject a `NgForm` instance and `NgbTooltip` instance
- subscribe to form control's validity change observable
- on each control's validity change you would have to open / close tooltip conditionally.

What I'm describing here can be seen in action in this plunker:
http://plnkr.co/edit/amoE4IyrfAwfOnWeKlt6?p=preview

The bottom line is that ng-bootstrap tooltips can be injected and used programatically. This is a proper way to go about form validation and tooltips.

Now, even if all what I'm describing here works and you can use this approach today, I think that we can simplify it further and make it even easier on users. I'm going to keep this issue open as a feature request, so we can improve integration with form validation. In any case you should not be blocked here any more.

**Comment by bastienmoulia** [comment]
@xyrintech If you want to use tooltips for form validation, you can already use `.{valid|invalid}-tooltip`. 
It's not well documented actually but it will be with the stable version of Bootstrap 4:  https://github.com/twbs/bootstrap/pull/25143.

Minimal plnkr: http://plnkr.co/edit/6DUKnswtOiJVQRV2hotn?p=preview

**Comment by pkozlowski-opensource** [comment]
After pondering this a bit more I think that solution proposed in https://github.com/ng-bootstrap/ng-bootstrap/issues/2033#issuecomment-353569393 is the most flexible one. Doing a dedicated control on ng-bootstrap side would not bring lots value but would limit flexibility.

Closing for now - what is requested here can be _very_ easily achieved in the user land.
